src/index.ts

```diff
@@ -1,3 +1,101 @@
-export default async function myFunction(input: any) {
-    return "Hello world!\n";
+import axios from "axios";
+import URI from "urijs";
+import isUrl from "is-url";
+import { Record } from "@magda/registry-client";
+
+export type UrlProcessorResult = {
+    dataset: Record;
+    distributions: Record[];
+};
+
+const isNumber = (v: any) => typeof v === "number";
+
+function getSpatialCoverageAspect(data: any) {
+    if (
+        !isNumber(data?.extent?.xmax) ||
+        !isNumber(data?.extent?.ymax) ||
+        !isNumber(data?.extent?.xmin) ||
+        !isNumber(data?.extent?.ymin)
+    ) {
+        return null;
+    }
+    return {
+        // -- Bounding box in order minlon (west), minlat (south), maxlon (east), maxlat (north)
+        bbox: [
+            data.extent.xmin,
+            data.extent.ymin,
+            data.extent.xmax,
+            data.extent.ymax
+        ],
+        spatialDataInputMethod: "bbox"
+    };
+}
+
+export default async function myFunction(
```

Comment:
Maybe a better name than `myFunction`? ;)

Reply:
will rename it to `processEsriApiUrl` --- if you have a better name, just let me know